samples/imports/import_extern.jakt

```diff
@@ -0,0 +1,51 @@
+/// Expect: selfhost-only
+/// - output: "PASS\n"
+
+import extern c "stdlib.h"  {
```

Comment:
A C++ test would also be nice, though it doesn't actually work as it is - here's the test I had with my branch:
```
import extern "LibJS/Interpreter.h" {
    namespace AK {
        extern struct NonnullRefPtr<T> {
            function ptr(this) -> raw T
        }
        extern struct NonnullOwnPtr<T> {
            function ptr(this) -> raw T
        }
        extern struct StringView {
            function StringView(anon s: raw c_char) -> StringView
        }
        extern struct Vector<T> {
            function size(this) -> usize
            function at(this, anon index: usize) -> T
        }
        extern struct Result<T, E> {
            function is_error(this) -> bool
            function release_value(mut this) -> T
        }
    }
    namespace JS {
        namespace Parser {
            extern struct Error {
            }
        }
        extern struct GlobalObject {
        }
        extern struct Value {
            function as_i32(this) -> i32
        }
        extern struct VM {
            // This is a lie, this takes default parameters.
            function create() -> AK::NonnullRefPtr<JS::VM>
        }
        extern struct ThrowCompletionOr<T> {
            function is_error(this) -> bool
            function release_value(mut this) -> T
        }
        extern struct Realm {
            function create(anon vm: JS::VM) -> raw JS::Realm
        }
        extern struct Script {
            function parse(source: AK::StringView, anon realm: JS::Realm) -> AK::Result<AK::NonnullRefPtr<JS::Script>, AK::Vector<JS::Parser::Error> >
        }
        extern struct Interpreter {
            function create<GlobalObjectType>(anon vm: JS::VM) -> AK::NonnullOwnPtr<JS::Interpreter>
            // This is a lie, this is not const.
            function run(this, anon script: JS::Script) -> JS::ThrowCompletionOr<JS::Value>
        }
    }
}

function main() {
    unsafe {
        mut vm = JS::VM::create();
        mut realm = JS::Realm::create(*vm.ptr());
        let src = "42 + 27;"
        mut result = JS::Script::parse(source: AK::StringView(src.c_string()), *realm);
        if result.is_error() {
            return 1
        }

        mut script = result.release_value();
        mut interpreter = JS::Interpreter::create<JS::GlobalObject>(*vm.ptr());
        mut exec_result = (*interpreter.ptr()).run(*script.ptr());
        if exec_result.is_error() {
            return 2
        }

        let value = exec_result.release_value();
        println("{}", value.as_i32())
    }
}
```

Reply:
That will need extending Jakttest to accept extra flags, and make sure we have serenity/lagom & js setup in CI to add that as a test. Otherwise it will fail with an include/link error from clang++

Reply:
Right, that one will also need a serenity build, perhaps we could just use a regular STL thing to test this?
e.g. 
```
import extern "vector" {
    namespace std {
        extern struct vector<T> {
             extern function push_back(mut this, anon value: T)
             extern function back(this) -> T
        }
    }
}

function main() {
     let v = std::vector<i32>()
     v.push_back(69i32)
     if v.back() == 69i32 {
        println("PASS")
     }
}
```

Reply:
nice, this triggered a couple of errors. Fixing those

Reply:
Seems like using STL confuses clang++ about forward().

Reply:
Got it to work by just initializing an empty vec and checking it's empty.